test/test.js

```diff
@@ -502,6 +504,117 @@ var betweennessCentrality = function (input, directed, node, centrality) {
+    var sheet = xlsx.parse(input);
+    var network = parseAdditionalSheet(sheet);
+    var additionalSheetMissingColumnHeaderErrorCount = 0;
+    for (let page in network.test) {
+        additionalSheetMissingColumnHeaderErrorCount += network.test[page].errors.filter(function (x) {
+            return x.errorCode === "MISSING_COLUMN_HEADER";
+        }).length;
+    }
+    assert.equal(frequency, additionalSheetMissingColumnHeaderErrorCount);
+};
+
+var twoColumnInvalidGeneTypeError = function (input, frequency) {
+    var sheet = xlsx.parse(input);
+    var network = parseAdditionalSheet(sheet);
+    var twoColumnInvalidGeneTypeErrorCount = 0;
+    for (let page in network.test) {
+        twoColumnInvalidGeneTypeErrorCount += network.test[page].errors.filter(function (x) {
+            return x.errorCode === "INVALID_GENE_TYPE";
+        }).length;
+    }
+    assert.equal(frequency, twoColumnInvalidGeneTypeErrorCount);
+};
+
+var twoColumnInvalidValueError = function (input, frequency) {
+    var sheet = xlsx.parse(input);
+    var network = parseAdditionalSheet(sheet);
+    var twoColumnInvalidValueErrorCount = 0;
+    for (let page in network.test) {
+        twoColumnInvalidValueErrorCount  += network.test[page].errors.filter(function (x) {
+            return x.errorCode === "INVALID_VALUE";
+        }).length;
+    }
+    assert.equal(frequency, twoColumnInvalidValueErrorCount);
+};
+
+var twoColumnInvalidGeneLengthError = function (input, frequency) {
+    var sheet = xlsx.parse(input);
+    var network = parseAdditionalSheet(sheet);
+    var twoColumnInvalidGeneLengthErrorCount = 0;
+    for (let page in network.test) {
+        twoColumnInvalidGeneLengthErrorCount  += network.test[page].errors.filter(function (x) {
+            return x.errorCode === "INVALID_GENE_LENGTH";
+        }).length;
+    }
+    assert.equal(frequency, twoColumnInvalidGeneLengthErrorCount);
+};
+
+var twoColumnSpecialCharacterError = function (input, frequency) {
+    var sheet = xlsx.parse(input);
+    var network = parseAdditionalSheet(sheet);
+    var twoColumnSpecialCharacterErrorCount = 0;
+    for (let page in network.test) {
+        twoColumnSpecialCharacterErrorCount += network.test[page].errors.filter(function (x) {
+            return x.errorCode === "INVALID_CHARACTER";
+        }).length;
+    }
+    assert.equal(frequency, twoColumnSpecialCharacterErrorCount);
+};
+
+// New Warning Tests
+
+var additionalSheetExtraneousDataWarning = function (input, frequency) {
+    var sheet = xlsx.parse(input);
+    var network = parseAdditionalSheet(sheet);
+    var additionalSheetExtraneousDataWarningCount = 0;
+    for (let page in network.test) {
+        additionalSheetExtraneousDataWarningCount  += network.test[page].warnings.filter(function (x) {
+            return x.warningCode === "EXTRANEOUS_DATA";
+        }).length;
+    }
+    assert.equal(frequency, additionalSheetExtraneousDataWarningCount);
+};
+
+
+
+
+
+
+
```

Comment:
Hmmm, if you really wanted to separate things this distinctly, maybe a new module file is called for? 🤔